r/tests/testthat/test-csv.R

```diff
@@ -541,3 +541,21 @@ test_that("write_csv_arrow can write from arrow_dplyr_query objects", {
   expect_named(tbl_in, c("dbl", "lgl", "false", "chr"))
   expect_equal(nrow(tbl_in), 3)
 })
+
+test_that("write_csv_arrow can write from RecordBatchReader objects", {
+  skip_if_not_available("dataset")
+  library(dplyr, warn.conflicts = FALSE)
+
+  query_obj <- arrow_table(tbl_no_dates) %>%
+    filter(lgl == TRUE)
+
+  rb_reader <- Scanner$create(query_obj)$ToRecordBatchReader()
+
+  csv_file <- tempfile()
+  tbl_out <- write_csv_arrow(rb_reader, csv_file)
```

Comment:
Pass in the query object to test that code path (it will get turned into an RBR in the function). Also be sure to clean up the tempfile.

```suggestion
  csv_file <- tempfile()
  on.exit(unlink(csv_file)
  tbl_out <- write_csv_arrow(query_obj, csv_file)
```